access/views.py

```diff
@@ -39,10 +46,125 @@ def index(request):
+    if store_version_id_path.exists():
+        with open(store_version_id_path) as f:
+            store_version_id = f.read()
+
+    if store_version_id == request.POST.get("version_id"):
+        renames([
+            (store_version_id_path, version_id_path),
+            (store_course_path, course_path),
+        ])
+
+        return HttpResponse()
+
+    version_id = None
+    if version_id_path.exists():
+        with open(version_id_path) as f:
+            version_id = f.read()
+
+    if version_id == request.POST.get("version_id"):
+        return HttpResponse()
+    else:
+        return HttpResponse("Unknown version id", status=400)
+
+
+@login_required
+def configure(request):
+    '''
+    Configure a course according to the gitmanager protocol.
+    '''
+    if request.method != "POST":
+        raise Http404()
+
+    if "publish" in request.POST and request.POST["publish"]:
+        return publish(request)
+
+    if "exercises" not in request.POST or "course_id" not in request.POST:
+        return HttpResponse("Missing exercises or course_id", status=400)
+
+    exercises = json.loads(request.POST["exercises"])
+    course_id = request.POST["course_id"]
+
+    if not request.auth.permissions.instances.has(Permission.WRITE, id=int(course_id)):
+        return HttpResponse(status=401)
+
+    root_dir = Path(settings.COURSE_STORE)
+    course_path = root_dir / course_id
+    if course_path.exists():
+        rmtree(course_path)
+
+    course_files_path = course_path / "files"
+    course_exercises_path = course_path / "exercises"
+    version_id_path = root_dir / (course_id + ".version")
+    course_files_path.mkdir(parents=True, exist_ok=True)
+    course_exercises_path.mkdir(parents=True, exist_ok=True)
+
+    if "files" in request.FILES:
+        tar_file = request.FILES["files"].file
+        tarh = TarFile(fileobj=tar_file)
+        tarh.extractall(course_files_path)
+
+    course_config = {
+        "name": course_id,
+        "exercises": [ex["key"] for ex in exercises],
+        "exercise_loader": "access.config._ext_exercise_loader",
+    }
+
+    with open(course_path / "index.json", "w") as f:
+        json.dump(course_config, f)
+
+    for info in exercises:
+        with open(course_exercises_path / (info["key"] + ".json"), "w") as f:
+            json.dump(info["config"], f)
+
+    if "version_id" in request.POST:
+        with open(version_id_path, "w") as f:
+            f.write(request.POST["version_id"])
+    elif version_id_path.exists():
+        version_id_path.unlink()
+
+    course_config = config._course_root_from_root_dir(course_id, root_dir)
+
+    defaults = {}
+    for info in exercises:
+        of = info["spec"]
+        if info.get("config"):
+            of["config"] = info["key"] + ".json"
+            course, exercise = config.exercise_entry(course_config, info["key"], "_root")
+            of = export.exercise(request, course, exercise, of)
+        defaults[of["key"]] = of
```

Comment:
It wasn't clear to me why all the configs are exported to the JsonResponse. Does the new Git manager use them? (Does the Git manager call this configure view?)

Reply:
Yes, gitmanager calls this. The response is the default (not specified in course configuration) values for the exercise configs passed to A+.